packages/datagateway-dataview/src/views/card/cardView.component.tsx

```diff
@@ -0,0 +1,814 @@
+
+  // Pagination.
+  const [page, setPage] = React.useState(-1);
+  const [dataCount, setDataCount] = React.useState(-1);
+  const [numPages, setNumPages] = React.useState(-1);
+  const [maxResults, setMaxResults] = React.useState(-1);
+  const [pageChange, setPageChange] = React.useState(false);
+  // By default, show pagination component at bottom of card view.
+  const paginationPos = paginationPosition ? paginationPosition : 'both';
+
+  // Change in data.
+  const [filterChange, setFilterChange] = React.useState(false);
+  const [sortChange, setSortChange] = React.useState(false);
+
+  // Filters.
+  const hasInfoFilters = information
+    ? information.some((i) => i.filterComponent)
+    : false;
+
+  const [filtersInfo, setFiltersInfo] = React.useState<CVFilterInfo>({});
+  const [selectedFilters, setSelectedFilters] = React.useState<
+    CVSelectedFilter[]
+  >([]);
+
+  // Sort.
+  const [cardSort, setCardSort] = React.useState<CVSort[] | null>(null);
+  const hasSort =
+    !title.disableSort ||
+    (description ? !description.disableSort : false) ||
+    (information ? information.some((i) => !i.disableSort) : false);
+
+  React.useEffect(() => {
+    // Get sort information from title, description and information lists.
+    let sortList: CVSort[] = [];
+
+    if (!title.disableSort) {
+      sortList.push({
+        label: title.label ? title.label : title.dataKey,
+        dataKey: title.dataKey,
+      });
+    }
+
+    if (description && !description.disableSort) {
+      sortList.push({
+        label: description.label ? description.label : description.dataKey,
+        dataKey: description.dataKey,
+      });
+    }
+
+    if (information) {
+      sortList = sortList.concat(
+        information
+          .filter((i) => !i.disableSort)
+          .map((i) => ({
+            label: i.label ? i.label : i.dataKey,
+            dataKey: i.dataKey,
+          }))
+      );
+    }
+
+    setCardSort(sortList);
+  }, [description, information, title.dataKey, title.disableSort, title.label]);
+
+  // Set the filter information based on what was provided.
+  React.useEffect(() => {
+    const getSelectedFilter = (
+      filterKey: string,
+      filterValue: string
+    ): boolean => {
+      if (filterKey in filters) {
+        const v = filters[filterKey];
+        if (Array.isArray(v) && v.includes(filterValue)) {
+          return true;
+        }
+      }
+      return false;
+    };
+
+    // Get the updated info.
+    const info = customFilters
+      ? Object.values(customFilters).reduce((o, filter) => {
+          const data: CVFilterInfo = {
+            ...o,
+            [filter.dataKey]: {
+              label: filter.label,
+              items: filter.filterItems.reduce(
+                (o, item) => ({
+                  ...o,
+                  [item]: getSelectedFilter(filter.dataKey, item),
+                }),
+                {}
+              ),
+              // TODO: Make use of selected items
```

Comment:
@patrick-austin this could be a separate issue since the idea here was to allow for the filter panel to open with the selected items already selected. At the moment due to the multiple renders going on, the item gets selected in the first render but any subsequent renders does not have the filter panel open.